server/server.go

```diff
@@ -114,3 +132,16 @@ func searchFor(field string, items []string) []string {
 func changeExt(pathname string, ext string) string {
 	return strings.TrimSuffix(pathname, path.Ext(pathname)) + ext
 }
+
+func apiKeyMiddleware(key string, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if key == "" || r.Header.Get("x-API-key") != key {
+			w.WriteHeader(http.StatusForbidden)
+			writeJSON(w, apiError{"forbidden"})
+			return
+		}
+		mutex.Lock()
```

Comment:
This needs a better name, like apiLock, and it shouldn't be global. Put it on the server that the API is a part of.